data/World/Water Temple.json

```diff
@@ -114,39 +121,42 @@
         "locations": {
             "Water Temple Central Pillar Chest": "
                 can_use(Iron_Boots) and can_use(Zora_Tunic) and can_use(Hookshot) and 
-                ((Small_Key_Water_Temple, 6) or can_use(Bow) or can_use(Dins_Fire))",
-            "Water Temple GS Central Pillar": "
-                ((can_use(Longshot) or (can_use(Farores_Wind) and can_use(Hookshot))) and
-                 ((Small_Key_Water_Temple, 6) or can_use(Bow) or can_use(Dins_Fire))) or
-                (Child_Water_Temple and Boomerang and can_use(Farores_Wind) and
-                    (can_use(Dins_Fire) or Sticks))"
+                ((Small_Key_Water_Temple, 6) or can_use(Bow) or can_use(Dins_Fire))"
         },
         "exits": {
             "Water Temple Cracked Wall": "True"
         }
     },
+    {
+        "region_name": "Water Temple Falling Platform Room",
+        "dungeon": "Water Temple",
+        "locations": {
+            "Water Temple GS Falling Platform Room": "
+                can_use(Longshot) or
+                (logic_water_falling_platform_gs_hookshot and can_use(Hookshot)) or
+                (logic_water_falling_platform_gs_boomerang and can_use(Boomerang))"
+        },
+        "exits": {
+            "Water Temple Dark Link Region": "(Small_Key_Water_Temple, 6) and can_use(Hookshot)"
```

Comment:
The falling platform room used to be in dark link region, but I had to separate it off when giving child access in order to boomerang the token. The exit to that room is still 5 keys. Exiting from that room to the rest of the dark link region there was no reason to keep the 6 keys checks on the locations themselves, since now there's a region entirely behind the 6-key door.